runner/src/main.rs

```diff
@@ -331,40 +342,54 @@ struct Client {
     additional_args: Vec<String>,
 }
 
-fn run_example(example: &Example) -> Step {
+fn run_example(opt: &RunExamples, example: &Example) -> Step {
     Step::Multiple {
         name: example.name.to_string(),
         steps: vec![
-            Step::Multiple {
-                name: "build_wasm_modules".to_string(),
-                steps: example
-                    .modules
-                    .iter()
-                    .map(|(name, target)| match target {
-                        Target::Cargo { cargo_manifest } => {
-                            build_wasm_module(name, &cargo_manifest)
-                        }
-                        Target::Bazel { .. } => todo!(),
-                    })
-                    .collect(),
-            },
-            build_example_config(&example.name),
-            Step::WithBackground {
-                name: "run_server".to_string(),
-                background: run_example_server(&format!(
-                    "./bazel-bin/examples/{}/config/config.bin",
-                    example.name
-                )),
-                foreground: Box::new(Step::Multiple {
-                    name: "run_client".to_string(),
+            vec![
+                Step::Multiple {
+                    name: "build_wasm_modules".to_string(),
                     steps: example
-                        .clients
+                        .modules
                         .iter()
-                        .map(|(name, client)| run_client(name, &client))
+                        .map(|(name, target)| match target {
+                            Target::Cargo { cargo_manifest } => {
+                                build_wasm_module(name, &cargo_manifest)
+                            }
+                            Target::Bazel { .. } => todo!(),
+                        })
                         .collect(),
+                },
+                build_example_config(&example.name),
+                // Build the server first so that when running it in the next step it will start up
+                // faster.
+                build_server(&BuildServer {
+                    variant: "rust".to_string(),
                 }),
+            ],
+            if opt.build_only {
```

Comment:
Passing the whole `RunExamples` struct as arg, only to use a single key seems a bit heavy. That's since the `application_variant ` and `example_name` keys are still present but not relevant. 

Could be a bit misleading to, if `opt.example_name` were ever set to a different name than `example.name` :).

Doesn't seem ideal from an API perspective, but I'm not super picky about it, since this is an internal function and not exposed as part of the runner API.

Reply:
(I'm aware due to the current implementation of L140, the misleading case I outlined never happens, but imho "the func API doesn't allow for misleading args" > "The func is currently never called in a misleading way")

Reply:
Let me think some more about it, perhaps I can split the struct in two and solve both this and your other point. Will follow up with a separate PR.